### PR TITLE
chore(chart): rename doc ref of `controller.securityContext` to `agent.securityContext`

### LINKED ISSUE #5757: Helm value `controller.securityContext` is documented but not used
### What's wrong?

The Helm value `controller.securityContext` is documented in [operations/helm/charts/grafana-agent/README.md](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md) and [operations/helm/charts/grafana-agent/README.md.gotmpl](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md.gotmpl) in multiple sections but a) missing in the [README.md Values section](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md#values) and b) not referenced in any file of the [operations/helm/](https://github.com/grafana/agent/tree/main/operations/helm) directory beside the README files.

I therefore speculate that the Helm value `controller.securityContext` got renamed, e.g. to `agent.securityContext`.

### Steps to reproduce

Find the Helm value `controller.securityContext` documented in [operations/helm/charts/grafana-agent/README.md](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md) and [operations/helm/charts/grafana-agent/README.md.gotmpl](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md.gotmpl) files but a) missing in the [README.md Values section](https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/README.md#values) and b) not referenced in any file of the [operations/helm/](https://github.com/grafana/agent/tree/main/operations/helm) directory beside the README files.

### System information

_No response_

### Software version

_No response_

### Configuration

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Fixes #5757 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added

### CODE REVIEW

**Comment by clayton-cornell** [comment]
@grafana/grafana-agent-maintainers Can one of you take a quick look and verify this change before I get the merge happening.

**Review by rfratto — APPROVED** [review]
Thanks @hainenber!

**Pull request merged**
